src/components/slide-toggle/slide-toggle.scss

```diff
@@ -0,0 +1,132 @@
+@import "../../core/style/variables";
+@import "../../core/style/mixins";
+@import "../../core/style/elevation";
+
+//TODO(): remove the default theme.
+@import "../../core/style/default-theme";
+
+$md-slide-toggle-width: 36px !default;
+$md-slide-toggle-height: 24px !default;
+$md-slide-toggle-bar-height: 14px !default;
+$md-slide-toggle-thumb-size: 20px !default;
+$md-slide-toggle-margin: 16px !default;
+
+:host {
+  display: flex;
+  height: $md-slide-toggle-height;
+
+  margin: $md-slide-toggle-margin 0;
+  line-height: $md-slide-toggle-height;
+
+  white-space: nowrap;
+  cursor: pointer;
+  user-select: none;
+
+  outline: none;
+
+  &[disabled] {
+    cursor: default;
+
+    .md-container {
+      cursor: default;
+    }
+  }
+
+  .md-container {
+    cursor: grab;
+    width: $md-slide-toggle-width;
+    height: $md-slide-toggle-height;
+
+    position: relative;
+    user-select: none;
+
+    margin-right: 8px;
+  }
+
+  .md-thumb-container {
+    position: absolute;
+    top: $md-slide-toggle-height / 2 - $md-slide-toggle-thumb-size / 2;
+    left: 0;
+    z-index: 1;
+
+    width: $md-slide-toggle-width - $md-slide-toggle-thumb-size;
+
+    transform: translate3d(0, 0, 0);
+
+    transition: $swift-linear;
+    transition-property: transform;
+
+    .md-thumb {
+      position: absolute;
+      margin: 0;
+      left: 0;
+      top: 0;
+
+      height: $md-slide-toggle-thumb-size;
+      width: $md-slide-toggle-thumb-size;
+      border-radius: 50%;
+
+      background-color: md-color($md-background, background);
+      @include md-elevation(1);
+    }
+  }
+
+  &.md-checked .md-thumb-container {
+    transform: translate3d(100%, 0, 0);
+  }
+
+  &.md-dragging .md-container {
+    cursor: grabbing;
+  }
+
+  .md-bar {
+    position: absolute;
+    left: 1px;
+    top: $md-slide-toggle-height / 2 - $md-slide-toggle-bar-height / 2;
+
+    width: $md-slide-toggle-width - 2px;
+    height: $md-slide-toggle-bar-height;
+
+    background-color: md-color($md-grey, 500);
+
+    border-radius: 8px;
+  }
+
+  .md-bar,
+  .md-thumb {
+    transition: $swift-linear;
+    transition-property: background-color;
+    transition-delay: 0.05s;
+  }
+
+  @mixin md-switch-checked($palette) {
+    .md-thumb {
+      background-color: md-color($palette);
+    }
+
+    .md-bar {
+      background-color: md-color($palette, 0.5);
+    }
+  }
+
+  &.md-checked {
+    @include md-switch-checked($md-accent);
+
+    &.md-primary {
+      @include md-switch-checked($md-primary);
+    }
+
+    &.md-warn {
+      @include md-switch-checked($md-warn);
+    }
+  }
+
+  &[disabled] {
+    .md-thumb {
+      background-color: md-color($md-grey, 400);

```

Comment:
This one shouldn't reference `$md-grey` directly, and instead use the theme's background palette.